tests/MongoDB.Driver.Core.Tests/Core/Clusters/MultiServerClusterTests.cs

```diff
@@ -671,9 +1147,32 @@ public void Should_call_dispose_on_servers_when_the_cluster_is_disposed()
             _capturedEvents.Any().Should().BeFalse();
         }
 
-        private MultiServerCluster CreateSubject()
+        // private methods
+        private void AssertTryGetEventHandlerWasCalled<TEvent>(Mock<IEventSubscriber> mockEventSubscriber)
+        {
+            Action<TEvent> handler;
+            mockEventSubscriber.Verify(m => m.TryGetEventHandler<TEvent>(out handler), Times.Once);
+        }
+
+        private Mock<IDnsMonitorFactory> CreateMockDnsMonitorFactory()
+        {
+            var mockDnsMonitorFactory = new Mock<IDnsMonitorFactory>();
+            mockDnsMonitorFactory
+                .Setup(m => m.CreateDnsMonitor(It.IsAny<IDnsMonitoringCluster>(), It.IsAny<string>(), It.IsAny<CancellationToken>()))
+                .Returns(Mock.Of<IDnsMonitor>());
```

Comment:
Just curious: how does Mock work for IDnsMonitor.Start?  It can return a mock Thread?

Reply:
The mock version of IDnsMonitor will return null when Start is called (because we didn't set it up to do anything else).

But these tests do not require that the mock IDnsMonitor actually be functional in any significant way. In fact, they actually require that there NOT be a functioning IDnsMonitor because we are going to manually feed dns events to this class by calling PublishDnsResults and PublishDnsException.

Curious you should ask, because I had a very hard time getting these tests to pass until I realized that I was getting real DNS events mixed in with my published ones, causing the DNS related tests to fail.

Reply:
OK, I see.  The MultiServerCluster never does anything with the Thread but store it in a field.  That's a bit weird in itself, but I think you said elsewhere that it lets you unit test something.  I'm not sure I get that, but it's also not that important either.

Reply:
OK, I see now.  It's a bit unexpected for MultiServerCluster to store a field that it doesn't use just to make it unit-testable, but it's not important enough now to change.  Maybe just make a note of it in an inline comment?